spec/services/monthly_district_data_service_spec.rb

```diff
@@ -24,49 +39,75 @@ def find_in_csv(csv_data, row_index, column_name)
-      hypertensive_patient
+      missed_visit_patient
+      patient_without_hypertension
+      ltfu_patient
+
       result = service.report
       csv = CSV.parse(result)
       facility_row_index = 4
+
       expect(find_in_csv(csv, facility_row_index, "#")).to eq("1")
       expect(find_in_csv(csv, facility_row_index, "District")).to eq(region.name)
-      expect(find_in_csv(csv, facility_row_index, "Facility")).to eq(facility.name)
-      expect(find_in_csv(csv, facility_row_index, "Block")).to eq(facility.block)
+      expect(find_in_csv(csv, facility_row_index, "Facility")).to eq(facility1.name)
+      expect(find_in_csv(csv, facility_row_index, "Block")).to eq(facility1.block)
       expect(find_in_csv(csv, facility_row_index, "Active/Inactive (Inactive facilities have 0 BP measures taken)"))
         .to eq("Inactive")
       expect(find_in_csv(csv, facility_row_index, "Estimated hypertensive population")).to eq(nil)
-      expect(find_in_csv(csv, facility_row_index, "Total registrations")).to eq("0")
-      expect(find_in_csv(csv, facility_row_index, "Total assigned patients")).to eq("1")
-      expect(find_in_csv(csv, facility_row_index, "Lost to follow-up patients")).to eq("0")
-      expect(find_in_csv(csv, facility_row_index, "Died all-time")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Total registrations")).to eq("2")
+      expect(find_in_csv(csv, facility_row_index, "Total assigned patients")).to eq("2")
+      expect(find_in_csv(csv, facility_row_index, "Lost to follow-up patients")).to eq("1")
+      dead = find_in_csv(csv, facility_row_index, "Dead patients (All-time as of #{Date.current.strftime("%e-%b-%Y")})")
+      expect(dead).to eq("0")
       expect(find_in_csv(csv, facility_row_index, "Patients under care")).to eq("1")
-      expect(csv[facility_row_index][12..24].uniq).to eq(["0"])
+      new_registrations = csv[facility_row_index][12..17]
+      expect(new_registrations).to eq(%w[0 0 1 0 0 0])
+      follow_ups = csv[facility_row_index][18..23]
+      expect(follow_ups).to eq(%w[0 0 0 0 0 0])
       expect(find_in_csv(csv, facility_row_index, "Patients with BP controlled")).to eq("0")
       expect(find_in_csv(csv, facility_row_index, "Patients with BP not controlled")).to eq("0")
       expect(find_in_csv(csv, facility_row_index, "Patients with a missed visits")).to eq("0")
-      expect(find_in_csv(csv, facility_row_index, "Patients with a visit but no BP taken")).to eq("0")
+      expect(find_in_csv(csv, facility_row_index, "Patients with a visit but no BP taken")).to eq("1")
       expect(csv[facility_row_index][30..31].uniq).to eq([nil])
     end
+
+    it "scopes the report to the provided period" do
+      old_period = Period.month("July 2018")
+      result = described_class.new(region, old_period).report
+      csv = CSV.parse(result)
+      column_headers = csv[2]
+      expect(column_headers[12]).to eq("Feb-2018")
+      expect(column_headers[17]).to eq("Jul-2018")
```

Comment:
Is there any significance to the magic numbers 12 and 17 here?  It may be more clear here to do something like this:

```
  first_month_index = 12
  last_month_index = 17
  expect(column_headers[first_month_index]).to eq("Feb-2018")
  expect(column_headers[last_month_index]).to eq("Feb-2018")
```



  
  column_headers[month_header_start_index].upto(months_of_date).with_index { |